server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -1566,7 +1566,7 @@ public UserVm removeNicFromVirtualMachine(RemoveNicFromVMCmd cmd) throws Invalid
         }
 
         // check to see if nic is attached to VM
-        if (nic.getInstanceId() != vmId) {
+        if (!nic.getInstanceId().equals(vmId)) {
```

Comment:
Is it possible for `nic.getInstanceId()` to be `null`? If it can be `null`, this will throw an NPE.

Reply:
```suggestion
        if (nic.getInstanceId() == null || !nic.getInstanceId().equals(vmId)) {
```